openmeter/entitlement/adapter/entitlement.go

```diff
@@ -529,6 +582,71 @@ func (a *entitlementDBAdapter) ListNamespacesWithActiveEntitlements(ctx context.
 	)
 }
 
+func (a *entitlementDBAdapter) GetScheduledEntitlements(ctx context.Context, namespace string, subjectKey models.SubjectKey, featureKey string, starting time.Time) ([]entitlement.Entitlement, error) {
+	res, err := entutils.TransactingRepo(
+		ctx,
+		a,
+		func(ctx context.Context, repo *entitlementDBAdapter) (*[]entitlement.Entitlement, error) {
+			res, err := repo.db.Entitlement.Query().
+				Where(
+					db_entitlement.Or(
+						db_entitlement.ActiveToIsNil(),
+						db_entitlement.ActiveToGT(starting),
+					),
+				).
+				Where(
+					db_entitlement.Or(db_entitlement.DeletedAtIsNil(), db_entitlement.DeletedAtGT(clock.Now())),
```

Comment:
Don't we need to also filter for ActiveTo < clock.Now()?

Reply:
I suppose you mean `activeFrom < clock.Now()`

What this query tries to do is simply get all entitlements that are relevant now or will be relevant later
(scheduled to activate later, or are currently active)

There are some data consistency presumptions with this query, being that no two can be active at the same time and `activeFrom` < `activeTo`